src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -88,33 +97,76 @@ public FileHashStore(HashMap<String, Object> hashstoreProperties)
                     + storeWidth;
             logFileHashStore.fatal(errMsg);
             throw new IllegalArgumentException(errMsg);
         }
+
         // Ensure algorithm supplied is not empty, not null and supported
         this.validateAlgorithm(storeAlgorithm);
 
-        // If no path provided, create default path with user.dir root + /FileHashStore
-        if (storePath == null) {
-            logFileHashStore.debug("FileHashStore - storePath is null, creating store root with default setitngs.");
-            String rootDirectory = System.getProperty("user.dir");
-            String defaultPath = "FileHashStore";
-            this.OBJECT_STORE_DIRECTORY = Paths.get(rootDirectory).resolve(defaultPath).resolve("objects");
+        // Check to see if configuration exists before initializing
+        Path hashstoreYamlPredictedPath = Paths.get(storePath + "/hashstore.yaml");
+        if (Files.exists(hashstoreYamlPredictedPath)) {
+            logFileHashStore.debug("FileHashStore - 'hashstore.yaml' found, verifying properties.");
+
+            HashMap<String, Object> hsProperties = this.getHashStoreYaml(storePath);
+            Path existingStorePath = (Path) hsProperties.get("storePath");
+            int existingStoreDepth = (int) hsProperties.get("storeDepth");
+            int existingStoreWidth = (int) hsProperties.get("storeWidth");
+            String existingStoreAlgorithm = (String) hsProperties.get("storeAlgorithm");
+
+            // Verify properties when 'hashstore.yaml' found
+            checkConfigurationEquality("store path", storePath, existingStorePath);
+            checkConfigurationEquality("store depth", storeDepth, existingStoreDepth);
+            checkConfigurationEquality("store width", storeWidth, existingStoreWidth);
+            checkConfigurationEquality("store algorithm", storeAlgorithm, existingStoreAlgorithm);
+
         } else {
-            this.OBJECT_STORE_DIRECTORY = storePath.resolve("objects");
+            // Check if HashStore exists at the given store path (and is missing config)
+            logFileHashStore.debug(
+                    "FileHashStore - 'hashstore.yaml' not found, check store path for objects and directories");
+
+            if (Files.exists(storePath)) {
+                boolean hashStoreExists = false;
+                try (Stream<Path> walk = Files.walk(storePath)) {
+                    if (walk.anyMatch(Files::exists)) {
+                        hashStoreExists = true;
+                    }
+                }
+                if (hashStoreExists) {
+                    String errMsg = "FileHashStore - Missing 'hashstore.yaml' but HashStore directories and/or objects found";
+                    logFileHashStore.fatal(errMsg);
+                    throw new IllegalStateException();
+                }
+            }
+            logFileHashStore.debug(
+                    "FileHashStore - 'hashstore.yaml' not found and store path not yet initialized.");
+
         }
```

Comment:
@artntek When you free up from your tasks, can you please help me double check this code block? I was experiencing some mvn test issues with it. Please ping me on Slack if you need any clarification - thank you! (Line 142 - 161)

Reply:
This looks fine to me, with the `throw` being done outside the try-with-resources block. 

However, I'd like to understand your criterion for deciding whether the hashstore already exists. From your code, it seems you're using the presence of at least one file in the subtree below the `storePath` directory, to determine existence. 

Does this mean that if you found a hashstore-style directory tree which has no files in it, you'd be OK concluding the hashstore does not exist?

I'm asking because the solution could be much simpler if you only wanted to determine whether anything at all (files or directories) exists in the `storePath` directory:

`storePath.toFile().listFiles()` would give you an array of its contents (both dirs and files). No need to walk the tree.

Reply:
@artntek Your suggestion is actually how I wanted to determine whether HashStore exists or not.
- When we arrive at this point in the code, `hashstore.yaml` is not present, but if directories/objects are found, my thoughts are that a store of some sort exists.
- In this case, I think it's better for HashStore to throw an exception to force the calling app to change the store path so as not to cause any conflict with what's already there (or do some "manual labour" to copy/write a valid `hashstore.yaml` to allow HashStore to continue there).

**RE: Implementation**
```
try (Stream<Path> walk = Files.walk(storePath)) {
    if (walk.anyMatch(Files::exists)) {
        hashStoreExists = true;
    }
}
```

Here is how I expect this try block to operate:
- The try block creates a lazily populated stream of paths (`Stream<Path> walk`) using `Files.walk(storePath)`.
- The [anyMatch()](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-) method is called on the stream, and it will return true as soon as the condition/predicate inside (Files::exists) is met for any path in the stream.
    - Files::exists checks for the existence of [directories and files, _not just files_](https://www.baeldung.com/java-file-directory-exists#nio).
- If the first path in the stream exists, the `anyMatch()` method will short-circuit the test, set hashStoreExists to true, and immediately move on to the next if statement to throw the exception.
- If my understanding here is correct, I won't actually need to walk over the entire tree or have to count all directories/files.

If I were to use `storePath.toFile().listFiles()` to get an array of its contents, is my assumption right that we would count the array (and if >0, set the boolean flag to true)? Your suggestion does seem simpler. I thought what I had was the right way to proceed, but now I'm not so sure....

Reply:
exactly - although you could also get rid of the boolean flag:
```
    File[] contents = storePath.toFile().listFiles();
    if (contents == null || contents.length > 0) {
        // throw the exception
    }
```
Your solution also works and is elegant, performant and modern, but seems more complex than needed for this particular task, especially since it runs only once, so performance isn't a primary concern.

The simpler solution is also quicker and easier to grok (thinking about our future selves - [we spend way more time reading other people's code than writing our own](https://skeptics.stackexchange.com/questions/48560/is-code-read-more-often-than-its-written#:~:text=Indeed%2C%20the%20ratio%20of%20time,makes%20it%20easier%20to%20write.))

Reply:
I recommend also going back and checking everywhere you've used `Files.exists()` -- I believe it returns true if it finds a file OR a directory, and in some cases, your code should distinguish between the two. Note there's also a `Files.isDirectory()` method that might be helpful (eg line 146)

Reply:
Sounds good! Thank you for the feedback and discussion - I will move forward with your suggestion (and also go back and check my usage for `Files.exists()`).